### PR TITLE
Add values source settings to template tags

### PR DESCRIPTION
**Pull request by ranquild**
Epic https://github.com/metabase/metabase/issues/26898

Adds a way to set a custom source for string template tags. Currently has no effect on the filter wiget.

How to test:
- New -> SQL query
- `SELECT * FROM PRODUCTS WHERE {{f}}`
- The template sidebar should appear with `How should users filter on this variable?` section
- Make sure it is possible to set the dropdown source for the tag

Screenshot 2023-01-17 at 15 44 24

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/27713)

### CODE REVIEW

File: frontend/src/metabase/parameters/components/ParameterLinkedFilters/ParameterLinkedFilters.tsx [hunk 1]
```diff
@@ -111,12 +116,12 @@ const ParameterLinkedFilters = ({
 };
 
 interface LinkedParameterProps {
-  parameter: UiParameter;
```
Comment (ranquild) [inline frontend/src/metabase/parameters/components/ParameterLinkedFilters/ParameterLinkedFilters.tsx:114]: `UiParameter` is dashboard-specific. Cards have `ParameterWithTarget`. Both extend `Parameter`.

File: frontend/src/metabase/parameters/components/ParameterLinkedFilters/use-filter-fields.ts [hunk 2]
```diff
@@ -40,18 +40,10 @@ const useFilterFields = (
   return state;
 };
 
-const getParameterError = ({ name }: UiParameter) => {
+const getParameterError = ({ name }: Parameter) => {
   return t`To view this, ${name} must be connected to at least one field.`;
 };
 
-const getParameterFieldIds = (parameter: UiParameter) => {
```
Comment (ranquild) [inline frontend/src/metabase/parameters/components/ParameterLinkedFilters/use-filter-fields.ts:47]: Remove duplication here and use the existing utility function.

File: frontend/test/metabase/scenarios/dashboard-filters/dashboard-filters-source.cy.spec.js [hunk 3]
```diff
@@ -90,113 +95,12 @@ describe("scenarios > dashboard > filters", () => {
...
     setFilter("Text or Category", "Is");
     mapFilterToQuestion();
-    editDropdown();
-    setupCustomList();
+    setFilterListSource({ values: ["Doohickey", "Gadget"] });
     saveDashboard();
     filterDashboard();
   });
-
-  it("should result in a warning being shown when archiving a question it uses", () => {
```
Comment (gusaiani) [inline frontend/test/metabase/scenarios/dashboard-filters/dashboard-filters-source.cy.spec.js:99]: Are we simply deleting this test?

File: frontend/test/metabase/scenarios/dashboard-filters/dashboard-filters-source.cy.spec.js [hunk 4]
```diff
@@ -217,3 +121,11 @@ const filterDashboard = () => {
     cy.wait("@getCardQuery");
   });
 };
+
+const archiveQuestion = () => {
+  openQuestionActions();
+  cy.findByTestId("archive-button").click();
```
Comment (kulyk) [inline frontend/test/metabase/scenarios/dashboard-filters/dashboard-filters-source.cy.spec.js:127]: Can we use `findByRole` here?

**Review by kulyk — APPROVED** [review]
LGTM

File: frontend/test/metabase/scenarios/dashboard-filters/dashboard-filters-source.cy.spec.js (hunk 3 above)
Comment (gusaiani) [reply frontend/test/metabase/scenarios/dashboard-filters/dashboard-filters-source.cy.spec.js:99]: No. Redistributed.

**Review by gusaiani — APPROVED** [review]
👍

**Pull request merged**